pkg/config/staging.go

```diff
@@ -113,6 +113,46 @@ func (bp *Blueprint) evalGhpcStageInModuleSettings() error {
 	return errs.OrNil()
 }
 
+// Update terraform_providers settings in place, evaluating `ghpc_stage` expressions
+func (bp *Blueprint) evalGhpcStageInTerraformProviderConfiguration() error {
+	errs := Errors{}
+	ctx, err := bp.evalCtx()
+	if err != nil {
+		return err
+	}
+	// Evaluate default providers
+	for k1, v1 := range bp.TerraformProviders {
+		us := map[string]cty.Value{}
+		for k2, v2 := range v1.Configuration.Items() {
```

Comment:
A lot of code duplication, let's create a method `evalGhpcStageInDict(Dict) -> Dict` to reduce it.

Reply:
Done.